keras_nlp/backend/ops.py

```diff
@@ -22,3 +25,26 @@
     from keras_core.src.backend.tensorflow.math import *  # noqa: F403, F401
     from keras_core.src.backend.tensorflow.nn import *  # noqa: F403, F401
     from keras_core.src.backend.tensorflow.numpy import *  # noqa: F403, F401
+
+
+if keras_core.config.backend() == "tensorflow" or not multi_backend():
+
+    def take_along_axis(x, indices, axis=None):
```

Comment:
This is the one nasty workaround for the `ops` layer I have left. I don't really like it yet, so haven't moved it into core, but we should find a solution for a `gather` like op in keras-core that is more friendly to dynamic shapes in tf.

Without this, shapes are popping out of `take_along_axis` with fully unknown shapes (e.g. `None, None, None`), which is breaking other parts of `fit()` on our models.